hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -56,11 +56,14 @@ class ReadOnlyHandle
   /// Returns true if handle references a value.
   inline operator bool() const { return value_ptr_ != nullptr; }
 
-  const std::string & get_name() const { return name_; }
+  const std::string get_name() const { return name_ + "/" + interface_name_; }
 
   const std::string & get_interface_name() const { return interface_name_; }
 
-  const std::string get_full_name() const { return name_ + "/" + interface_name_; }
+  // Deprecated
+  const std::string get_full_name() const { return get_name(); }
+
+  const std::string & get_prefix_name() const { return name_; }
```

Comment:
Would it make sense to also rename this member to `prefix_name_` just to be clearer in the future?

```suggestion
  const std::string & get_prefix_name() const { return prefix_name_; }
```